include/neml2/tensors/R2.h

```diff
@@ -24,98 +24,35 @@
 
 #pragma once
 
-#include "neml2/tensors/FixedDimTensor.h"
+#include "neml2/tensors/R2Base.h"
 
 namespace neml2
 {
-class Scalar;
-class Vec;
-class SR2;
-class R3;
 class Rot;
+class SR2;
+class WR2;
 
 /**
- * @brief The (logical) full second order tensor.
+ * @brief A basic R2
  *
- * The logical storage space is (3, 3).
+ * The logical storage space is (3,3).
  */
-class R2 : public FixedDimTensor<R2, 3, 3>
+class R2 : public R2Base<R2>
```

Comment:
Should R2 be transformable?

Reply:
Right now it doesn't have to be, but if/when we need it to be it will be very easy to implement.  Just make it inherit from the ABC and implement `transform` in the usual way.  I'm not going to bother to do it and write the tests until we find that we need it.